### PR TITLE
CAMEL-14608: add support for dead letter policies to camel-pulsar

### PR DESCRIPTION
**Pull request by connormcauliffe-toast**
Addresses [CAMEL-14608](https://issues.apache.org/jira/browse/CAMEL-14608)

**Motivation**
Pulsar has support to add Dead Letter Topics to consumers for messages which have exceeded the max retry count to get sent to. Camel should be able to support the creation and customization of dead letter policies on Pulsar Endpoints.
Link to Pulsar documentation for Dead Letter Topics: https://pulsar.apache.org/docs/en/next/concepts-messaging/#dead-letter-topic

**Verifying the Changes**
Tests have been added to `PulsarConsumerDeadLetterPolicyTest` to verify the existence of the Dead Letter Topic and that it has been configured correctly according to the configuration given.

### CODE REVIEW

File: components/camel-pulsar/src/main/java/org/apache/camel/component/pulsar/configuration/PulsarConfiguration.java [hunk 1]
```diff
@@ -55,6 +55,10 @@
     private long ackGroupTimeMillis = 100;
     @UriParam(label = "consumer", defaultValue = "LATEST")
     private SubscriptionInitialPosition subscriptionInitialPosition = LATEST;
+    @UriParam(label = "consumer", description = "Maximum number of times that a message will be redelivered before being sent to the dead letter queue. If this value is not set, no Dead Letter Policy will be created")
+    private Integer maxRedeliverCount;
+    @UriParam(label = "consumer", description = "Name of the topic where the messages which fail {maxRedeliverCount} times will be sent. Note: if not set, default topic name will be <topicname>-<subscriptionname>-DLQ")
```
Comment (davsclaus) [inline components/camel-pulsar/src/main/java/org/apache/camel/component/pulsar/configuration/PulsarConfiguration.java]: Lets avoid using {xxx} in the description but just the name of the option is fine

Comment (davsclaus) [reply components/camel-pulsar/src/main/java/org/apache/camel/component/pulsar/configuration/PulsarConfiguration.java]: Also avoid the < > in the DLQ name as the generated docs in the json will strip out invalid characters.

File: components/camel-pulsar/src/main/java/org/apache/camel/component/pulsar/configuration/PulsarConfiguration.java [hunk 2]
```diff
@@ -354,4 +358,28 @@ public void setMessageRouter(MessageRouter messageRouter) {
         this.messageRouter = messageRouter;
     }
 
+    public Integer getMaxRedeliverCount() {
+        return maxRedeliverCount;
+    }
+
+    /**
```
Comment (davsclaus) [inline components/camel-pulsar/src/main/java/org/apache/camel/component/pulsar/configuration/PulsarConfiguration.java]: This javadoc and the option description are not the same, eg if the other options dont have javadoc then remove it also here, eg they are documented in the json/adoc file which is the important place

**Comment by davsclaus** [comment]
Looks great, only a few comments related to documenting the options and using simpler text to make it valid for json/adoc too

**Comment by oscerd** [comment]
I'll wait for the fix suggested here

**Comment by oscerd** [comment]
Once we merge CAMEL-14607 we need to rebase this.

**Comment by oscerd** [comment]
@connormcauliffe-toast can you please rebase and regen?

**Comment by connormcauliffe-toast** [comment]
@oscerd Rebase complete

**Pull request merged**